internal/gcb/gcb.go

```diff
@@ -29,22 +29,43 @@ import (
 // Client interface abstracts Cloud Build service interactions.
 type Client interface {
 	CreateBuild(ctx context.Context, project string, build *cloudbuild.Build) (*cloudbuild.Operation, error)
-	GetOperation(ctx context.Context, op *cloudbuild.Operation) (*cloudbuild.Operation, error)
+	WaitForOperation(ctx context.Context, op *cloudbuild.Operation) (*cloudbuild.Operation, error)
 }
 
-// Service is a concrete implementation using the Cloud Build service.
-type Service struct {
-	Service *cloudbuild.Service
+// clientImpl is a concrete implementation of the Client interface using the Cloud Build service.
+type clientImpl struct {
+	service      *cloudbuild.Service
+	pollInterval time.Duration
+}
+
+// NewClient creates a new Client with the given options.
+func NewClient(s *cloudbuild.Service) Client {
+	return &clientImpl{
+		service:      s,
+		pollInterval: 10 * time.Second,
```

Comment:
Want to add a comment here about why we chose 10 seconds? I remember GCB was rate limiting us at one point but I don't remember exactly how we came to 10 seconds.

Reply:
i think i did the math based on GCB's default quota but I'd like to apply for a quota bump so I won't be too specific in the comment.